src/System.Runtime.Extensions/tests/System/Convert.netcoreapp.cs

```diff
@@ -90,72 +94,258 @@ public void TryToBase64Chars_InvalidOptions_Throws(Base64FormattingOptions inval
         }
 
         [Theory]
-        [InlineData("")]
-        [InlineData("BQYHCA==")]
-        [InlineData(
-            "AAECAwQFBgcICQoLDA0ODxAREhMUFRYXGBkaGxwdHh8gISIjJCUmJygpKissLS4vMDEyMzQ1Njc4\r\n" +
-            "OTo7PD0+P0BBQkNERUZHSElKS0xNTk9QUVJTVFVWV1hZWltcXV5fYGFiY2RlZmdoaWprbG1ub3Bx\r\n" +
-            "cnN0dXZ3")]
-        public void TryFromBase64String_MatchesFromBase64String(string stringInput)
+        [MemberData(nameof(Base64TestData))]
+        public static void TryFromBase64String(string encoded, byte[] expected)
         {
-            byte[] expected = Convert.FromBase64String(stringInput);
-            Span<byte> dest;
+            if (expected == null)
+            {
+                Span<byte> actual = new byte[1000];
+                bool success = Convert.TryFromBase64String(encoded, actual, out int bytesWritten);
+                Assert.False(success);
+                Assert.Equal(0, bytesWritten);
+            }
+            else
+            {
+                // Exact-sized buffer
+                {
+                    byte[] actual = new byte[expected.Length];
```

Comment:
nit:
`Span<byte> actual = new byte[expected.Length];`, here and elsewhere. It isn't necessary though since we only have a single overload of Convert.TryFromBase64String and it takes a Span.

Reply:
It saves us the bother of having to construct a new array from the Span to pass to Assert.Equals().

Reply:
Good point!